### PR TITLE
Task1 progress

### PR DESCRIPTION
**Pull request by GhiseTeodora**

### CODE REVIEW

File: task1/force-app/main/default/classes/AccountContactListController.cls [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+	public Account account;
+	private Id primaryId;
+	public String searchText{ get; set; }
+	private List<Contact> contacts;
+    public Integer offset = 0;  
+	public Integer numberOfContactsOnThePage = 20; 
+    public integer totalNumber;
+	public AccountContactListController(ApexPages.StandardController controller){
+		this.account = (Account)controller.getRecord();
+		if (searchText == null){
```
[the remark this answers is not in the document]
Comment (GhiseTeodora) [reply task1/force-app/main/default/classes/AccountContactListController.cls:25]: Fixed

File: task1/force-app/main/default/classes/AccountContactListController.cls [hunk 2]
```diff
@@ -0,0 +1,245 @@
+
+/*
+* @author Teodora Ghise
+* @date July 13, 2021
+* @name searchController
+* @description controller for AccountPrimaryContact page 
+*/
+
+public with sharing class AccountContactListController{
+	public Account account;
```
[the remark this answers is not in the document]
Comment (GhiseTeodora) [reply task1/force-app/main/default/classes/AccountContactListController.cls]: fixed

File: task1/force-app/main/default/classes/AccountContactListController.cls [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+	private Id primaryId;
+	public String searchText{ get; set; }
+	private List<Contact> contacts;
+    public Integer offset = 0;  
+	public Integer numberOfContactsOnThePage = 20;
```
[the remark this answers is not in the document]
Comment (GhiseTeodora) [reply task1/force-app/main/default/classes/AccountContactListController.cls]: fixed

File: task1/force-app/main/default/classes/AccountContactListController.cls (hunk 1 above)
[the remark this answers is not in the document]
Comment (GhiseTeodora) [reply task1/force-app/main/default/classes/AccountContactListController.cls:25]: fixed

File: task1/force-app/main/default/classes/AccountContactListController.cls [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+			List<Contact> currentPrimaryContacts = securityDecision.getRecords();
+			for (Contact item : currentPrimaryContacts){
+				item.Is_Primary_Contact__c = false;
+			}
+			//clear existing primary contacts 
+			update currentPrimaryContacts;
+
+			//set the new contact as primary
+			Contact newPrimaryInstance = new Contact(Id = contactId, Is_Primary_Contact__c = true);
+			update newPrimaryInstance;
```
[the remark this answers is not in the document]
Comment (GhiseTeodora) [reply task1/force-app/main/default/classes/AccountContactListController.cls]: fixed

File: task1/force-app/main/default/classes/AccountContactListControllerTest.cls [hunk 5]
```diff
@@ -0,0 +1,105 @@
+@isTest
+public class AccountContactListControllerTest{
+
```
[the remark this answers is not in the document]
Comment (GhiseTeodora) [reply task1/force-app/main/default/classes/AccountContactListControllerTest.cls]: fixed

**Review by romanbychkoosf — APPROVED** [review]

**Pull request merged**